cassandra/src/test/java/se/tre/freki/storage/cassandra/search/CassandraSearchPluginTest.java

```diff
@@ -43,12 +52,50 @@ public void tearDown() throws Exception {
   public void testNGram() throws Exception {
     LabelMeta meta = LabelMeta.create(CassandraLabelId.fromLong(23L), LabelType.METRIC, "jungle",
         "and its description", 0L);
+    LabelMeta meta2 = LabelMeta.create(CassandraLabelId.fromLong(22L), LabelType.METRIC, "junglar",
+        "and its description", 0L);
 
+    searchPlugin.indexLabelMeta(meta2);
     searchPlugin.indexLabelMeta(meta);
+    System.out.println("dada");
+  }
+
+  @Test
+  public void testLabelSearch() throws Exception {
+    searchPlugin.findLabels("hello");
+  }
+
+  @Test
+  public void testFetchLabelIdsForEmptyString() throws Exception {
+    NGramGenerator ngram = new NGramGenerator("");
+    List<ResultSetFuture> result = searchPlugin.fetchLabelIds(ngram);
+    assertNotNull(result);
```

Comment:
Testing what happens when there are no n-grams is a good test case but I think using the `NGramGenerator` here couples things too hard. Letting the `#fetchLabelIds` method accept a general `Iterator<String>` instead and use `ImmutableSet.of().iterator()` here is more appropriate I believe. It might also be appropriate to change the `assrtNotNull` to `assertTrue(result.isEmpty())`.